### ISSUE TITLE
Feedback on DVE

### ISSUE THREAD

**Issue by rod-glover** [pacificclimate/dash-dv-explorer#25]
This issue is a way to provide wide-ranging feedback on the DVE at the present date (Dec 23, 2020). It is not actually a single issue, but potentially several. It will need to be triaged and broken up into actionable issues by the dev team.

Overall, the app looks and feels good. Most of what I have to say here suggests changes or improvements, but that is not meant to impugn its overall excellence, only to make it more excellent. Also, I am just parachuting in here at the last moment and may not be aware of design requirements or constraints, so any given suggestion could be naive or unhelpful. Triage at will.

## Overall

1. Layout: Clean and clear. Very nice use of whitespace.
1. Navigation: Simple, clear.
2. Map: Good appearance and functionality.
1. Interaction: Feels a little sluggish, specifically the map updates. The delay is long enough that the experience is, "Hey, what happened? I don't see any change. Is something wrong?" (at least to a naive user).
   - If possible to speed up (I suspect not), great. 
   - If not, then suggest  that a spinner be introduced whenever the map is being updated, i.e., after the user changes something in the UI (e.g., variable) and before the map reflects this change.

### Map

1. Variable selector
   1. Suggest label ("Design Variable"?)
   2. Suggest limiting width of selector to equivalent of about 10 characters.
1.  HSM Reconstruction / CanRCM4 Ensemble Mean: Suggest this as a dropdown instead of switch. You can't see what your options are without actually choosing one.
1. Colourscale: Maybe suggest a dropdown or radio button instead of switch.
2. Colourbar range: Is it possible to make this take on only "nice" values (e.g., multiples of 0.5 deg for temperatures)? This would have to account for the variable's natural range, so it couldn't be a single increment/multiple for all unless you are astoundingly lucky.
3. Colourbar: Like colourbar range, maybe round the increments to "nice" values? This is always a PITA, but there may be some canned solution you can adopt to avoid writing tedious, error-prone code.

### Table

1. Will users be looking repeatedly for DV's at a specific location? If so, suggest some kind of filtering that will help them locate the correct row. Alphabetical order, as present, helps a lot, but a filter is nice. See [Wx Files app](http://docker-dev01.pcic.uvic.ca:30530/) for an example.

## Help/Documentation

This app seems quite intuitive to me, but the addition of a little Help or even external documentation, explaining its purpose and use seems like a good idea. If integrating it is too much effort, you could provide a hyperlink at the top of the screen that opens a Help/Doc page in a new window.

**Comment by nannau** [comment]
Thank you, @rod-glover, your feedback is always appreciated and these are important points. Especially from a front-end guru! Some of these tasks sound like low hanging fruit.

To me, it makes sense to pipe these into separate issues 
* Loading spinner (#26)
* Variable Selector Label (great idea) (#27)
* Radio/Dropdown Selector for Ambiguous Toggles (#28)
* Table Filter (#29)

And then a couple of comments for the record and for future developers who may have similar questions:

**Speed:**
This is the current profile for timing after updating HSM (at target resolution == slower)
![image](https://user-images.githubusercontent.com/10455520/103034913-e5b14b80-451a-11eb-8ce0-b5bd584fb963.png)

Dash uses JSON data which is slow. There may be some things in `dash_app.update_ds` that can be factored out that stay consistent between callbacks to speed things up.

**Colourscale**
This has been a bit of a PITA. Plotly doesn't have easy-to-access hex codes like matplotlib, and it's difficult to come up with nice and descriptive discrete (as opposed to continuous) colourbars without custom code. Right now I'm doing a computation that scales between the min and maximum of the reconstruction, and perhaps rounding the zmin and zmax (cmin and cmax) would help with getting "nice" values, then the stepsize between colours would automatically be nice as well - or at least as nice as the (cmax-cmin)/(N_colours + 1) . 

Thanks again, Rod!

**Comment by rod-glover** [comment]
I agree about "nice" values; certainly getting the endpoints nice is a good start. I don't think the colourscale has to be continuous; the 10 or so colours you use seem quite suitable to me. It's more a question of scaling and I think your ideas are a good start.
